Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -107,6 +115,12 @@ def sse4_2?
         sysctl_bool("hw.optional.sse4_2")
       end
 
+      # Note: this is more reliable than checking uname.
+      # `sysctl` returns the right answer even when running in Rosetta.
+      def arm64?
+        sysctl_bool("hw.optional.arm64")
```

Comment:
I’m worried that `def arm64?; sysctl_bool("hw.optional.arm64"); end` is wallpapering over an underlying bug. `def arm64?; arm? && is_64_bit?; end` should be correct on both Linux and macOS under both Rosetta and native.

The fundamental issue is I believe that we need both `Hardware::CPU` (the emulated CPU when emulated otherwise the native CPU) and `Hardware::NativeCPU`

Reply:
Put another way… with this PR under Rosetta `arm?` is false, `is_64_bit?` is true, `arm64?` is true, which is 🤔

Reply:
`Hardware::NativeCPU` may be over-engineered. How about something like…
```ruby
def native_type
         sysctl_bool("hw.optional.arm64") ? "arm64" : type
end
```